src/planner/macos/mod.rs

```diff
@@ -66,6 +73,24 @@ pub struct Macos {
     /// The root disk of the target
     #[cfg_attr(feature = "cli", clap(long, env = "NIX_INSTALLER_ROOT_DISK"))]
     pub root_disk: Option<String>,
+
+    /// On AWS, put the Nix Store volume on the EC2 instances' instance store volume.
+    ///
+    /// WARNING: Using the instance store volume means the machine must never be Stopped in AWS.
+    /// If the instance is Stopped, the instance store volume is erased, and the installation is broken.
+    /// The machine can be safely rebooted.
+    ///
+    /// Using the instance store volume bypasses the interactive "enable full disk access" step.
+    /// Without this flag, installations on macOS on EC2 will require manual, graphical intervention when first installed to grant Full Disk Access.
+    ///
+    /// Setting this option:
+    ///  * Requires passing --determinate due to complications of AWS's deployment of macOS.
+    ///  * Sets --root-disk to an auto-detected disk
+    #[cfg_attr(
+        feature = "cli",
+        clap(long, default_value = "false", requires = "determinate_nix")
+    )]
```

Comment:
We could skip this since it requires determinate nix, but I don't think we need to since its already effectively disabled